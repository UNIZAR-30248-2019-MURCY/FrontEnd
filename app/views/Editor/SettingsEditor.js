import React, {Component} from 'react';
import {
    View,
    StyleSheet,
    Image, AsyncStorage,
} from 'react-native'
import {Button, colors, Text} from 'react-native-elements';
import {NavigationActions, StackActions} from "react-navigation";
import {removeItem} from "../../services/AsyncStorage/remove";

export default class SettingScreen extends Component {

    constructor(props) {
        super(props);
        this.signOff = this.signOff.bind(this);
    }

    signOff() {
        removeItem('token')
            .then((r) => {
                if (r) {
                    const toWelcome = StackActions.reset({
                        index: 0,
                        actions: [NavigationActions.navigate({routeName: 'Welcome'})],
                    });
                    this.props.navigation.dispatch(toWelcome);
                }
            });
    }

    render() {
        return (
            <View style={styles.container}>
                <View style={styles.containerTitle}>
                    <Text h2>Settings</Text>
                </View>

                <View style={styles.containerSettings}>
                    <Button
                        className='player-button'
                        type="clear"
                        buttonStyle={styles.button}
                        title="Player Mode"
                        titleStyle={styles.buttonText}
                        onPress={() => {
                            const toPlayer = StackActions.reset({
                                index: 0,
                                actions: [NavigationActions.navigate({ routeName: 'Player' })],
                            });
                            this.props.navigation.dispatch(toPlayer);
                        }}/>
                </View>

                <View style={styles.buttonSO}>
                    <Button
                        className='signoff-button'
                        type="clear"
                        title="Sign off"
                        titleStyle={styles.buttonText}
                        onPress={() => {
                            this.signOff()
                        }}/>
                </View>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        //justifyContent: 'center',
        padding: 20
    },
    containerTitle: {
        marginTop: 50,
        padding: 20
    },

    containerSettings: {
        padding: 20
    },
    button: {
        marginBottom: 5,
    },
    buttonText: {
        fontSize: 20,
        color: 'grey'
    },
    buttonSO: {
        position: 'absolute',
        bottom: 0,
        marginBottom: 50,
    }

})
