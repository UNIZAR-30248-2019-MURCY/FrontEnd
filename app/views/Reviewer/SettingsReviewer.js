import React, {Component} from 'react';
import {
    View,
    StyleSheet,
    Image, AsyncStorage,
} from 'react-native'
import {Button, colors, Text} from 'react-native-elements';
import {NavigationActions, StackActions} from "react-navigation";
import {removeItem} from "../../services/AsyncStorage/remove";
import {retrieveItem} from "../../services/AsyncStorage/retrieve";

export default class SettingsReviewer extends Component {

    constructor(props) {
        super(props);
        this.signOff = this.signOff.bind(this);
        this.state = {
            editor: false,
            player: false
        }
    }

    signOff() {
        removeItem('token')
            .then((r) => {
                if (r) {
                    const toWelcome = StackActions.reset({
                        index: 0,
                        actions: [NavigationActions.navigate({routeName: 'Welcome'})],
                    });
                    this.props.navigation.dispatch(toWelcome);
                }
            });
    }

    componentDidMount() {
        retrieveItem('role')
            .then(data => {
                console.log(data)
                if(data.includes("USER")){
                    this.setState({player: true})
                }
                if(data.includes("EDITOR")){
                    this.setState({editor: true})
                }
            })
    }

    render() {

        let player = (
            this.state.player ?
                <Button
                    className='player-button'
                    type="clear"
                    buttonStyle={styles.button}
                    title="Player Mode"
                    titleStyle={styles.buttonText}
                    onPress={() => {
                        const toEditor = StackActions.reset({
                            index: 0,
                            actions: [NavigationActions.navigate({routeName: 'Player'})],
                        });
                        this.props.navigation.dispatch(toEditor);
                    }}/> :
                <View></View>
        );

        let editor = (
            this.state.editor ?
                <Button
                    className='editor-button'
                    type="clear"
                    buttonStyle={styles.button}
                    title="Editor Mode"
                    titleStyle={styles.buttonText}
                    onPress={() => {
                        const toEditor = StackActions.reset({
                            index: 0,
                            actions: [NavigationActions.navigate({routeName: 'Editor'})],
                        });
                        this.props.navigation.dispatch(toEditor);
                    }}/> :
                <View></View>
        );

        return (
            <View style={styles.container}>
                <View style={styles.containerTitle}>
                    <Text h2>Settings</Text>
                </View>

                <View style={styles.containerSettings}>
                    {player}
                    {editor}
                </View>

                <View style={styles.buttonSO}>
                    <Button
                        className='signoff-button'
                        type="clear"
                        title="Sign off"
                        titleStyle={styles.buttonText}
                        onPress={() => {
                            this.signOff()
                        }}/>
                </View>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        //justifyContent: 'center',
        padding: 20
    },
    containerTitle: {
        marginTop: 50,
        padding: 20
    },

    containerSettings: {
        padding: 20
    },
    button: {
        marginBottom: 5,
    },
    buttonText: {
        fontSize: 20,
        color: 'grey'
    },
    buttonSO: {
        position: 'absolute',
        bottom: 0,
        marginBottom: 50,
    }

})
