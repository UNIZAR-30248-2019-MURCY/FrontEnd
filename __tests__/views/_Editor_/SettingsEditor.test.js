import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import SettingsEditor from "../../../app/views/Editor/SettingsEditor";
import SettingsReviewer from "../../../app/views/Reviewer/SettingsReviewer";

describe('<SettingsEditor />', () => {
    it('SettingsEditor renders without crashing', () => {
        const rendered = renderer.create(<SettingsEditor />).toJSON();
        expect(rendered).toBeTruthy();
    });
    it('SettingsEditor test against snapshot', () => {
        const page = renderer.create(<SettingsEditor />).toJSON();
        expect(page).toMatchSnapshot();
    });
    it('should navigate to Player', () => {
        const navigationMock = { dispatch: jest.fn() };
        const wrapper = shallow(<SettingsEditor navigation={navigationMock} />);
        wrapper.find('.player-button').simulate('press');
        expect(navigationMock.dispatch.mock.calls.length).toEqual(1);
    })
    it('signOff', () => {
        let wrapper = shallow(<SettingsEditor/>);
        wrapper.instance().signOff = jest.fn();
        wrapper.update();
        wrapper.find('.signoff-button').simulate('press');
        expect(wrapper.instance().signOff).toBeCalledTimes(1);
    })
});
