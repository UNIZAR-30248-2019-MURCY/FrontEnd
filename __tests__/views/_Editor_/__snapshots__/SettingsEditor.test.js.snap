// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsEditor /> SettingsEditor test against snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "marginTop": 50,
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 42.5,
        }
      }
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      Settings
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 20,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 5,
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "grey",
                "fontSize": 20,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Player Mode
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "marginBottom": 50,
        "position": "absolute",
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "grey",
                "fontSize": 20,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Sign off
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
